jdiff/evaluators.py

```diff
@@ -51,41 +51,41 @@ def parameter_evaluator(values: Mapping, parameters: Mapping, mode: str) -> Dict
         Dictionary with all the items that have some value not matching the expectations from parameters
     """
     if not isinstance(values, list):
-        raise TypeError("Something went wrong during jmespath parsing. 'values' must be of type List.")
+        raise TypeError("'values' must be of type List.")
 
     result = {}
-    for value in values:
+    for index, value in enumerate(values):
         # value: {'7.7.7.7': {'peerAddress': '7.7.7.7', 'localAsn': '65130.1101', 'linkType': 'externals
         if not isinstance(value, dict):
-            raise TypeError(
-                "Something went wrong during jmespath parsing. ",
-                f"'value' ({value}) must be of type Dict, and it's {type(value)}",
-            )
+            raise TypeError(f"'value' ({value}) must be of type Dict, and it's {type(value)}")
 
         result_item = {}
 
-        # TODO: Why the 'value' dict has always ONE single element? we have to explain
-        # inner_key: '7.7.7.7'
-        inner_key = list(value.keys())[0]
-        # inner_value: [{'peerAddress': '7.7.7.7', 'localAsn': '65130.1101', 'linkType': 'externals'}]
-        inner_value = list(value.values())[0]
+        # When data has been normalized with $key$, get inner key and value
+        if len(value) == 1:
+            # inner_key: '7.7.7.7'
+            inner_key = list(value.keys())[0]
+            # inner_value: [{'peerAddress': '7.7.7.7', 'localAsn': '65130.1101', 'linkType': 'externals'}]
+            value = list(value.values())[0]
+        else:
+            inner_key = index
```

Comment:
what is the value in this case?

Reply:
An element from the top for loop, where values is a function argument.

Reply:
If data normalisation is detected when, reassign value from inner element.

Reply:
> Jmespath parsing is now optional.

I believe this answering to my question above (?)